### PR TITLE
Error pages

### PR DESCRIPTION
**Pull request by somme**
- handle errors 404, 500 etc.

### CODE REVIEW

File: src/frontend/view/view.ts [hunk 1]
```diff
@@ -121,52 +127,59 @@ const viewMainLayout = () => {
   return result.join('')
 }
 
+const viewDataBaseError = () => {
+  const html = []
+  let errorText = ''
+  switch (state.systemErrorText) {
+    case 'TIMEOUT':
+      errorText =
+        'The databse took too long to respond. This is usually due to too many query results. Please try a more precise query.'
```
Comment (JazzPote) [inline src/frontend/view/view.ts:136]:
```suggestion
        'The database took too long to respond. This is usually due to too many query results. Please try a more precise query.'
```

Comment (JazzPote) [reply src/frontend/view/view.ts:136]: Are timeouts always database errors? What if the server times out for another reason?

File: src/backend/middleware/errorMiddleware.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { Response, Request } from 'express'
+
+export const errorMiddleware = (
+  error: any,
+  req: Request,
+  res: Response
+): void => {
+  const status = error.response.status || 500
```
Comment (JazzPote) [inline src/backend/middleware/errorMiddleware.ts:8]:
```suggestion
  const status = error.response?.status || 500
```

File: src/frontend/view/view.ts [hunk 3]
```diff
@@ -121,52 +127,59 @@ const viewMainLayout = () => {
...
             </h1>
             <ul class="govuk-error-summary__list">
           `)
-        state.userErrors.forEach((userError) => {
-          switch (userError) {
-            case 'missingWhereToSearch':
-              html.push(`
+      state.userErrors.forEach((userError) => {
+        switch (userError) {
+          case 'missingWhereToSearch':
```
Comment (JazzPote) [inline src/frontend/view/view.ts:167]: Could we put these constants in an enum?

File: src/frontend/view/view.ts [hunk 4]
```diff
@@ -121,52 +127,59 @@ const viewMainLayout = () => {
...
               <li><a href="#search-scope-wrapper">You need to select a publishing application</a></li>`)
-              break
-            default:
-              console.log('unknown user error code:', userError)
-          }
-        })
-        html.push(`
+            break
+          default:
+            console.log('unknown user error code:', userError)
```
Comment (JazzPote) [inline src/frontend/view/view.ts:176]:
```suggestion
            console.log('Unknown user error code:', userError)
```

**Review by JazzPote — COMMENTED** [review]
Nice one, a different approach with its strengths.
Just a few minor comments.

File: src/frontend/view/view.ts (hunk 1 above)
Comment (somme) [reply src/frontend/view/view.ts:136]: not sure, maybe one for @nacnudus

File: src/frontend/view/view.ts (hunk 3 above)
Comment (somme) [reply src/frontend/view/view.ts]: added enums

**Review by JazzPote — APPROVED** [review]
lgtm

**Pull request merged**